### PR TITLE
Social network : closing feature 7580

### PR DESCRIPTION
**Pull request by NicolasEYSSERIC**
- adding feature "Remove from my contacts"
- deleting obsolete code and JSP pages
- enhancing plugin 'invitMe' and renaming it in 'relationship' which now manage invitation, cancellation, acceptation and remove relation
- enhancing REST service 'invitations'
- removing old JSON servlet now that all interactions are made with REST service
- fixing bugs on silverpeas-userZoom.js (using Promesse to get contacts and invitations)

### CODE REVIEW

File: core-library/src/main/java/org/silverpeas/core/admin/user/model/UserDetail.java [hunk 1]
```diff
@@ -724,27 +725,45 @@ public String getDurationOfCurrentSession() {
...
+    return getInvitation(userId, getId());
+  }
+
+  private Invitation getInvitation(String fromUserId, String toUserId) {
+    InvitationService invitationService = InvitationService.get();
+    Invitation invitation = null;
+    try {
+      invitation =
+          invitationService.getInvitation(Integer.parseInt(fromUserId), Integer.parseInt(toUserId));
+    } catch (Exception e) {
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/admin/user/model/UserDetail.java]: MAJOR Either log or rethrow this exception. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1166)

File: core-library/src/main/java/org/silverpeas/core/socialnetwork/invitation/AcceptationUserNotification.java [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  protected void performTemplateData(final String language, final Invitation resource,
+      final SilverpeasTemplate template) {
+    UserFull user = UserFull.getById(getSender());
+
+    template.setAttribute("senderUser", user);
+    template.setAttribute("userName", user.getDisplayedName());
+  }
+
+  @Override
+  protected void performNotificationResource(final String language, final Invitation resource,
```
Comment (silveruser) [inline core-library/src/main/java/org/silverpeas/core/socialnetwork/invitation/AcceptationUserNotification.java]: MAJOR Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1186)

File: core-web/src/main/java/org/silverpeas/core/webapi/socialnetwork/RelationResource.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+public class RelationResource extends RESTWebService {
+
+  @Inject
+  private RelationShipService relationShipService;
+
+  @DELETE
+  @Path("{userId}")
+  @Consumes(MediaType.APPLICATION_JSON)
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response deleteRelation(final @PathParam("userId") String userId) {
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/webapi/socialnetwork/RelationResource.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: core-web/src/main/java/org/silverpeas/core/webapi/socialnetwork/invitation/InvitationResource.java [hunk 4]
```diff
@@ -30,10 +30,16 @@
 
 import javax.enterprise.context.RequestScoped;
 import javax.inject.Inject;
+import javax.ws.rs.Consumes;
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/webapi/socialnetwork/invitation/InvitationResource.java]: MINOR Remove this unused import 'javax.ws.rs.Consumes'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

File: core-web/src/main/java/org/silverpeas/core/webapi/socialnetwork/invitation/InvitationResource.java [hunk 5]
```diff
@@ -71,6 +77,42 @@
     return asWebEntity(invitations, locatedAt(getUriInfo().getAbsolutePathBuilder()));
   }
 
+  @Path("{id}")
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public InvitationEntity getInvitation(final @PathParam("id") Integer id) {
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/webapi/socialnetwork/invitation/InvitationResource.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: core-web/src/main/java/org/silverpeas/core/webapi/socialnetwork/invitation/InvitationResource.java [hunk 6]
```diff
@@ -71,6 +77,42 @@
...
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response cancelInvitation(final @PathParam("id") Integer id) {
+    invitationService.ignoreInvitation(id);
+    return Response.ok().build();
+  }
+
+  @PUT
+  @Path("{id}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response acceptInvitation(final @PathParam("id") Integer id) {
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/webapi/socialnetwork/invitation/InvitationResource.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AModifiersOrderCheck)

File: core-web/src/main/java/org/silverpeas/core/webapi/socialnetwork/invitation/InvitationResource.java [hunk 7]
```diff
@@ -71,6 +77,42 @@
...
+  @Produces(MediaType.APPLICATION_JSON)
+  public InvitationEntity getInvitation(final @PathParam("id") Integer id) {
+    Invitation invitation = invitationService.getInvitation(id);
+    return asWebEntity(invitation, locatedAt(getUriInfo().getAbsolutePathBuilder()));
+  }
+
+  @DELETE
+  @Path("{id}")
+  @Produces(MediaType.APPLICATION_JSON)
+  public Response cancelInvitation(final @PathParam("id") Integer id) {
```
Comment (silveruser) [inline core-web/src/main/java/org/silverpeas/core/webapi/socialnetwork/invitation/InvitationResource.java]: MINOR Reorder the modifiers to comply with the Java Language Specification. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AModifiersOrderCheck)

**Comment by silveruser** [comment]
SonarQube analysis reported 8 issues
* MAJOR 2 major
* MINOR 6 minor

Watch the comments in this conversation to review them.

#### 1 extra issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. MINOR [InvitationEntity.java#L32](https://github.com/Silverpeas/Silverpeas-Core/blob/a4a4cf0cf177f114469bc8fff957bf245e42d170/core-web/src/main/java/org/silverpeas/core/webapi/socialnetwork/invitation/InvitationEntity.java#L32): Remove this unused import 'javax.validation.constraints.NotNull'. [rule](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by SilverYoCha** [comment]
Merged.
Closed manually because of manual git operations performed during integration.

**Pull request closed without merging**